core/src/test/scala/zio/jdbc/ZConnectionPoolSpec.scala

```diff
@@ -173,6 +173,35 @@ object ZConnectionPoolSpec extends ZIOSpecDefault {
             } yield assertTrue(invalidatedClosed)
           }
         } +
+        test("auto invalidate dead connections") {
+          def testPoolInvalid(
+            config: ZConnectionPoolConfig = ZConnectionPoolConfig(
+              1,
+              1,
+              ZConnectionPoolConfig.defaultRetryPolicy,
+              300.seconds
+            ) //Only one connection in pool
+          ) =
+            for { //Pool that contains some invalid connections
+              conns  <- Queue.unbounded[TestConnection]
```

Comment:
No problem! Yes, it can be built without this queue. In the case of `testPool` the connections taken from the queue are being used here https://github.com/zio/zio-jdbc/blob/main/core/src/test/scala/zio/jdbc/ZConnectionPoolSpec.scala#L184. In your case you don't need to have access to all available connections in the pool so you can skip creating the queue.

Reply:
Thank you for the help! 

After having seen the test code again, I believe the best way to approach the creation of `testPoolInvalid` is to use the already existing `testPool` function changing the default config parameter.

 I've pushed these new changes which I believe make the test much clearer.